pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/StringSchema.java

```diff
@@ -93,6 +102,22 @@ public String decode(byte[] bytes) {
         }
     }
 
+    public String decode(ByteBuf byteBuf) {
+        if (null == byteBuf) {
+            return null;
+        } else {
+            int size = byteBuf.readableBytes();
+            byte[] bytes = tmpBuffer.get();
+            if (size > bytes.length) {
+                bytes = new byte[size * 2];
+                tmpBuffer.set(bytes);
+            }
+            byteBuf.readBytes(bytes, 0, size);
+
+            return new String(bytes, 0, size, charset);
```

Comment:
I think we should create a new buffer for the `String`. because the pass-in byte[] will be referenced by the `String` object.

Reply:
new String(bytes, 0, size, charset);  I looked at the source code and did a test, I did't find where be referenced. Can you give me some hints?

Reply:
@congbobo184 you are right. it is my mistake.